libs/application/template-api-modules/src/lib/modules/templates/children-residence-change/children-residence-change.service.ts

```diff
@@ -0,0 +1,104 @@
+import { Injectable, Inject } from '@nestjs/common'
+import { TemplateApiModuleActionProps } from '../../../types'
+import { AwsService } from '@island.is/application/file-service'
+import {
+  SyslumennService,
+  Person,
+  Attachment,
+  PersonType,
+} from '@island.is/api/domains/syslumenn'
+import {
+  CRCApplication,
+  Override,
+} from 'libs/application/templates/children-residence-change/src/types'
+import { User } from '@island.is/api/domains/national-registry'
+
+export const PRESIGNED_BUCKET = 'PRESIGNED_BUCKET'
+
+type props = Override<
+  TemplateApiModuleActionProps,
+  { application: CRCApplication }
+>
+
+@Injectable()
+export class ChildrenResidenceChangeService {
+  constructor(
+    private readonly awsService: AwsService,
+    private readonly syslumennService: SyslumennService,
+    @Inject(PRESIGNED_BUCKET) private readonly presignedBucket: string,
+  ) {}
+
+  // TODO: Send email to parents
+  async submitApplication({ application }: props) {
+    const s3FileName = `children-residence-change/${application.id}.pdf`
+    const file = await this.awsService.getFile(this.presignedBucket, s3FileName)
+    const fileContent = file.Body?.toString('base64')
+    const nationalRegistry = (application.externalData.nationalRegistry
+      .data as unknown) as User
+
+    // TODO: Remove ternary for usemocks once we move mock data to externalData
+    const selectedChildren =
+      application.answers.useMocks === 'no'
+        ? application.externalData.childrenNationalRegistry.data.filter((c) =>
+            application.answers.selectChild.includes(c.name),
+          )
+        : application.answers.mockData.childrenNationalRegistry.data.filter(
+            (c) => application.answers.selectChild.includes(c.name),
+          )
+
+    if (!fileContent) {
+      throw new Error('File content was undefined')
+    }
+
+    const attachment: Attachment = {
+      name: `Lögheimilisbreyting-barns-${nationalRegistry.nationalId}.pdf`,
+      content: fileContent,
+    }
+
+    const parentA: Person = {
+      name: nationalRegistry.fullName,
+      ssn: nationalRegistry.nationalId,
+      phoneNumber: application.answers.parentA.phoneNumber,
```

Comment:
This is just nitpicking but we could deconstruct like this:
```
const { answers, externalData } = application
const { parentNationalRegistry } = externalData
```

And save a few characters 😄